corehq/util/metrics/__init__.py

```diff
@@ -185,7 +195,19 @@ def inner(*args, **kwargs):
     return inner
 
 
-def create_metrics_event(title, text, alert_type=ALERT_INFO, tags=None, aggregation_key=None):
+def create_metrics_event(title: str, text: str, alert_type: str = ALERT_INFO,
+                         tags: Dict[str, str] = None, aggregation_key: str = None):
```

Comment:
I'm not a fan of these type hints due to the verbosity. It makes the function signature much harder to read. Ones that are particularly obvious/not needed are `title`, `text`, and `attregation_key`, but really, none of it adds that much value IMO. I think it would be better to add type info in the docstring unless/until we have type hint validation on travis.

Reply:
I do get your point but if you're using an IDE that supports it then it is helpful. Also, if we start doing static type checking with something like mypy then surely you want either all or none.

I'll leave this as is for now and start a discussion in a doc which I'll post on slack.

Reply:
> Also, if we start doing static type checking with something like mypy then surely you want either all or none.

I'm not sure what you mean by "all or none."

Do you mean every function in our codebase should have a type hint on every parameter? I _certainly_ do not want that. I hope this is not what you meant.

Maybe you meant that any function having type hints should have them on all parameters? This doesn't seem immediately obvious or desirable to me either. I don't know the state of the art on static type checking in Python, but maybe that's a current requirement/limitation? I would be surprised by that since it's possible to add type hints on some parameters while omitting them from others.

Reply:
Let's take this to https://docs.google.com/document/d/1SxDhlyBJVBcn1Jc2GHtIXyDtv2BTBPxiIxoks0lhXGk/edit#